### PR TITLE
fix: added missing push and reminders related types 

### PR DESCRIPTION
**Pull request by vishalnarkhede**
## Changelog

- Fixed the typescript for push config
- Added missing types for `automod_threshold`
- Added missing types for `reminders `
- Move the api keys and secrets for typescript test to github secret

### CODE REVIEW

**Review by ferhatelmas — APPROVED** [review]

File: src/types.ts [hunk 1]
```diff
@@ -1956,7 +1962,7 @@ export type Policy = {
   owner?: boolean;
   priority?: number;
   resources?: string[];
-  roles?: string[];
+  roles?: string[] | null;
```
Comment (peterdeme) [inline src/types.ts:1965]: isn't `?` means it's nullable already?

File: .github/workflows/type.yml [hunk 2]
```diff
@@ -9,7 +9,10 @@ jobs:
   test:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: ./.github/actions/setup-node
+      uses: actions/checkout@v2
+      uses: ./.github/actions/setup-node
```
Comment (peterdeme) [inline .github/workflows/type.yml:13]: this syntay isn't valid

File: src/types.ts [hunk 3]
```diff
@@ -75,8 +75,10 @@ export type AppSettingsAPIResponse<StreamChatGenerics extends ExtendableGenerics
     channel_configs: Record<
       string,
       {
+        reminders: boolean;
```
Comment (peterdeme) [inline src/types.ts:78]: i think this fails prettier: the properties should be in an alphabetical order

File: src/types.ts [hunk 4]
```diff
@@ -1542,8 +1545,10 @@ export type ChannelConfigAutomodThresholds = null | {
 };
 
 export type ChannelConfigFields = {
+  reminders: boolean;
```
Comment (peterdeme) [inline src/types.ts:1548]: i think this fails prettier: the properties should be in an alphabetical order

File: src/types.ts (hunk 1 above)
Comment (AnatolyRugalev) [reply src/types.ts:1965]: `?` means `string[] | undefined` which is a bit different from `null`. So the result type here is `string[] undefined | null`

File: .github/workflows/type.yml (hunk 2 above)
Comment (vishalnarkhede) [reply .github/workflows/type.yml:13]: haha yeah I am trying to figure out whats wrong exactly

File: test/typescript/response-generators/client.js [hunk 5]
```diff
@@ -174,7 +174,11 @@ async function syncTeam() {
 		text: 'New Event?',
 		user: { id: user1 },
 	});
-	return await client.sync([channel.cid], new Date(Date.now() - 1000 * 60).toISOString());
+	const response = await client.sync([channel.cid], new Date(Date.now() - 1000 * 60).toISOString());
+
+	const serverClient = await utils.disableMultitenancy();
```
Comment (AnatolyRugalev) [inline test/typescript/response-generators/client.js:179]: This won't play well with concurrent runs

File: src/types.ts (hunk 1 above)
Comment (AnatolyRugalev) [reply src/types.ts:1965]: > yeah but does it make sense here?

That's a good question :) I don't think this should be necessary

Comment (vishalnarkhede) [reply src/types.ts:1965]: not to me ... but backend sent `null` in one of our test case

File: test/typescript/response-generators/client.js (hunk 5 above)
Comment (vishalnarkhede) [reply test/typescript/response-generators/client.js:179]: hmm good point, creating the separate app is the only way then ... for multi-tenancy!!

File: src/types.ts (hunk 1 above)
Comment (vishalnarkhede) [reply src/types.ts:1965]: I can remove it if we can ensure it never null from backend response

File: src/types.ts (hunk 3 above)
Comment (vishalnarkhede) [reply src/types.ts:78]: Its a required property ... they get higher priority over optional ones

File: src/types.ts [hunk 6]
```diff
@@ -98,6 +100,7 @@ export type AppSettingsAPIResponse<StreamChatGenerics extends ExtendableGenerics
         url_enrichment?: boolean;
       }
     >;
+    reminders_interval: number;
```
[the remark this answers is not in the document]
Comment (vishalnarkhede) [reply src/types.ts:103]: Same ... required property

File: src/types.ts (hunk 4 above)
Comment (vishalnarkhede) [reply src/types.ts:1548]: Nope ... required vs optional

File: test/typescript/response-generators/client.js (hunk 5 above)
Comment (vishalnarkhede) [reply test/typescript/response-generators/client.js:179]: ok just gonna setup different app

**Review by AnatolyRugalev — APPROVED** [review]

**Pull request merged**